ctms/acoustic_service.py

```diff
@@ -396,16 +396,24 @@ def _add_contact(self, list_id=None, sync_fields=None, columns=None):
                 self.context["main_status"] = status
                 self.context["main_duration_s"] = duration_s
 
-    def _insert_update_relational_table(self, table_name, rows):
-        if not rows:
-            return
+    def _replace_relational_table_data(self, table_name, email_id, rows):
         start_time = time.monotonic()
-        status = "success"
         table_id = self.relational_tables[table_name]
+        status = "success"
+        method = "delete_relational_table_data"
         try:
-            self.acoustic.insert_update_relational_table(
-                table_id=table_id, rows=rows
-            )  # Call to Acoustic
+            # Calls to Acoustic
+            self.acoustic.delete_relational_table_data(
+                table_id=table_id,
+                rows=[
+                    {"email_id": email_id},
+                ],
+            )
+            if rows:
+                method = "insert_update_relational_table"
+                self.acoustic.insert_update_relational_table(
+                    table_id=table_id, rows=rows
+                )
```

Comment:
So we delete all `newsletter_rt` data associated with `email_id`; then 
if newsletter data exists _("existing" newsletter data stored in CTMS)_ for `email_id` insert all data into `newsletter_rt`?

It looks like we'd be updating the timestamps (_`create/update`_) in the [newsletter_converter](https://github.com/mozilla-it/ctms-api/blob/11610ffca765f320d37412cceffbfef18795fba7/ctms/acoustic_service.py#L217) 
https://github.com/mozilla-it/ctms-api/blob/11610ffca765f320d37412cceffbfef18795fba7/ctms/acoustic_service.py#L240-L246

Reply:
~It looks like we'd be updating the timestamps (_`create/update`_) in the [newsletter_converter](https://github.com/mozilla-it/ctms-api/blob/11610ffca765f320d37412cceffbfef18795fba7/ctms/acoustic_service.py#L217) 
https://github.com/mozilla-it/ctms-api/blob/11610ffca765f320d37412cceffbfef18795fba7/ctms/acoustic_service.py#L240-L246~

_Misread the dict.get call. Ignore this!_